### ISSUE TITLE
Dateformat failure after upgrading to 2024.12.05

### ISSUE THREAD

**Issue by pvh0** [xirixiz/homeassistant-afvalwijzer#352]
### The problem

After updateing I get the below errors and the integration fails to work.

### Version of afvalwijzer integration having the issue?

2024.12.05

### Version of Home Assistant Core having the issue?

2024.12.5

### Anything in the logs that might be useful for us?

```txt
2024-12-20 14:43:49.369 ERROR (SyncWorker_5) [custom_components.afvalwijzer.const.const] Collector initialization failed: time data '2025-13-11' does not match format '%Y-%m-%d'

2024-12-20 14:43:49.465 ERROR (MainThread) [custom_components.afvalwijzer.const.const] Failed to fetch waste types: 'NoneType' object has no attribute 'keys'
```

### Additional information

Sensors are not available anymore "This entity is no longer being provided by the afvalwijzer integration. If the entity is no longer in use, delete it in settings."

**Comment by xirixiz** [comment]
Please, always add as much info as possible. What is your provider, etc, etc.
No problem for now, just saying ;).

Anyway, sounds you have a similar issue like here: https://github.com/xirixiz/homeassistant-afvalwijzer/issues/350

In which "gemeente" do you live?

**Comment by csmale** [comment]
I have the same. Mijnafvalwijzer is giving bad data in one field, but that is killing the whole integration. It would be better if just that one entry could be skipped.
Gemeente Stichtse Vecht. The API call returns the following. Note that all the dates are YYYY-MM-DD until you get towards the bottom when you come across dates as YYYY-DD-MM. The first bad date is 2025-13-11, which is what is shown in the log.

`{"ophaaldagen":{"response":"OK","data":[{"nameType":"papier en karton","type":"papier","date":"2024-01-05"},{"nameType":"gft","type":"gft","date":"2024-01-11"},{"nameType":"pbd","type":"pmd","date":"2024-01-17"},{"nameType":"papier en karton","type":"papier","date":"2024-01-26"},{"nameType":"pbd","type":"pmd","date":"2024-02-07"},{"nameType":"gft","type":"gft","date":"2024-02-08"},{"nameType":"papier en karton","type":"papier","date":"2024-02-22"},{"nameType":"pbd","type":"pmd","date":"2024-02-28"},{"nameType":"gft","type":"gft","date":"2024-03-07"},{"nameType":"pbd","type":"pmd","date":"2024-03-20"},{"nameType":"gft","type":"gft","date":"2024-03-21"},{"nameType":"papier en karton","type":"papier","date":"2024-03-29"},{"nameType":"gft","type":"gft","date":"2024-04-04"},{"nameType":"pbd","type":"pmd","date":"2024-04-10"},{"nameType":"gft","type":"gft","date":"2024-04-18"},{"nameType":"pbd","type":"pmd","date":"2024-05-01"},{"nameType":"gft","type":"gft","date":"2024-05-02"},{"nameType":"papier en karton","type":"papier","date":"2024-05-03"},{"nameType":"gft","type":"gft","date":"2024-05-16"},{"nameType":"pbd","type":"pmd","date":"2024-05-22"},{"nameType":"gft","type":"gft","date":"2024-05-30"},{"nameType":"papier en karton","type":"papier","date":"2024-05-31"},{"nameType":"pbd","type":"pmd","date":"2024-06-12"},{"nameType":"gft","type":"gft","date":"2024-06-13"},{"nameType":"gft","type":"gft","date":"2024-06-27"},{"nameType":"papier en karton","type":"papier","date":"2024-06-28"},{"nameType":"pbd","type":"pmd","date":"2024-07-03"},{"nameType":"gft","type":"gft","date":"2024-07-11"},{"nameType":"pbd","type":"pmd","date":"2024-07-24"},{"nameType":"gft","type":"gft","date":"2024-07-25"},{"nameType":"papier en karton","type":"papier","date":"2024-07-26"},{"nameType":"gft","type":"gft","date":"2024-08-08"},{"nameType":"pbd","type":"pmd","date":"2024-08-14"},{"nameType":"gft","type":"gft","date":"2024-08-22"},{"nameType":"papier en karton","type":"papier","date":"2024-08-30"},{"nameType":"pbd","type":"pmd","date":"2024-09-04"},{"nameType":"gft","type":"gft","date":"2024-09-05"},{"nameType":"gft","type":"gft","date":"2024-09-19"},{"nameType":"pbd","type":"pmd","date":"2024-09-25"},{"nameType":"papier en karton","type":"papier","date":"2024-09-27"},{"nameType":"gft","type":"gft","date":"2024-10-03"},{"nameType":"pbd","type":"pmd","date":"2024-10-16"},{"nameType":"gft","type":"gft","date":"2024-10-17"},{"nameType":"papier en karton","type":"papier","date":"2024-10-25"},{"nameType":"gft","type":"gft","date":"2024-10-31"},{"nameType":"pbd","type":"pmd","date":"2024-11-06"},{"nameType":"gft","type":"gft","date":"2024-11-14"},{"nameType":"pbd","type":"pmd","date":"2024-11-27"},{"nameType":"gft","type":"gft","date":"2024-11-28"},{"nameType":"papier en karton","type":"papier","date":"2024-11-29"},{"nameType":"gft","type":"gft","date":"2024-12-12"},{"nameType":"pbd","type":"pmd","date":"2024-12-18"},{"nameType":"papier en karton","type":"papier","date":"2024-12-28"}],"error":false},"ophaaldagenNext":{"response":"OK","data":[{"nameType":"grof tuinafval","type":"tuinafval","date":"2025-01-04"},{"nameType":"gft","type":"gft","date":"2025-01-09"},{"nameType":"gft","type":"gft","date":"2025-01-23"},{"nameType":"grof tuinafval","type":"tuinafval","date":"2025-01-24"},{"nameType":"pbd","type":"pmd","date":"2025-01-28"},{"nameType":"gft","type":"gft","date":"2025-02-06"},{"nameType":"papier en karton","type":"papier","date":"2025-02-10"},{"nameType":"grof tuinafval","type":"tuinafval","date":"2025-02-14"},{"nameType":"gft","type":"gft","date":"2025-02-20"},{"nameType":"pbd","type":"pmd","date":"2025-02-25"},{"nameType":"gft","type":"gft","date":"2025-03-06"},{"nameType":"grof tuinafval","type":"tuinafval","date":"2025-03-07"},{"nameType":"gft","type":"gft","date":"2025-03-20"},{"nameType":"pbd","type":"pmd","date":"2025-03-25"},{"nameType":"grof tuinafval","type":"tuinafval","date":"2025-03-28"},{"nameType":"gft","type":"gft","date":"2025-04-03"},{"nameType":"papier en karton","type":"papier","date":"2025-04-09"},{"nameType":"gft","type":"gft","date":"2025-04-17"},{"nameType":"grof tuinafval","type":"tuinafval","date":"2025-04-18"},{"nameType":"pbd","type":"pmd","date":"2025-04-22"},{"nameType":"gft","type":"gft","date":"2025-05-01"},{"nameType":"grof tuinafval","type":"tuinafval","date":"2025-05-09"},{"nameType":"gft","type":"gft","date":"2025-05-15"},{"nameType":"pbd","type":"pmd","date":"2025-05-20"},{"nameType":"grof tuinafval","type":"tuinafval","date":"2025-05-30"},{"nameType":"gft","type":"gft","date":"2025-05-31"},{"nameType":"pbd","type":"pmd","date":"2025-06-17"},{"nameType":"grof tuinafval","type":"tuinafval","date":"2025-06-20"},{"nameType":"papier en karton","type":"papier","date":"2025-07-08"},{"nameType":"grof tuinafval","type":"tuinafval","date":"2025-07-11"},{"nameType":"pbd","type":"pmd","date":"2025-07-15"},{"nameType":"grof tuinafval","type":"tuinafval","date":"2025-08-01"},{"nameType":"pbd","type":"pmd","date":"2025-08-12"},{"nameType":"grof tuinafval","type":"tuinafval","date":"2025-08-22"},{"nameType":"pbd","type":"pmd","date":"2025-09-09"},{"nameType":"grof tuinafval","type":"tuinafval","date":"2025-09-12"},{"nameType":"grof tuinafval","type":"tuinafval","date":"2025-10-03"},{"nameType":"papier en karton","type":"papier","date":"2025-10-07"},{"nameType":"pbd","type":"pmd","date":"2025-10-07"},{"nameType":"grof tuinafval","type":"tuinafval","date":"2025-10-24"},{"nameType":"pbd","type":"pmd","date":"2025-11-04"},{"nameType":"papier en karton","type":"papier","date":"2025-11-12"},{"nameType":"grof tuinafval","type":"tuinafval","date":"2025-11-14"},{"nameType":"pbd","type":"pmd","date":"2025-12-02"},{"nameType":"grof tuinafval","type":"tuinafval","date":"2025-12-05"},{"nameType":"papier en karton","type":"papier","date":"2025-12-06"},{"nameType":"grof tuinafval","type":"tuinafval","date":"2025-12-24"},{"nameType":"pbd","type":"pmd","date":"2025-12-30"},{"nameType":"papier en karton","type":"papier","date":"2025-13-11"},{"nameType":"papier en karton","type":"papier","date":"2025-16-10"},{"nameType":"papier en karton","type":"papier","date":"2025-18-10"},{"nameType":"papier en karton","type":"papier","date":"2025-21-09"},{"nameType":"papier en karton","type":"papier","date":"2025-24-08"},{"nameType":"papier en karton","type":"papier","date":"2025-26-07"},{"nameType":"papier en karton","type":"papier","date":"2025-27-12"},{"nameType":"papier en karton","type":"papier","date":"2025-29-"},{"nameType":"papier en karton","type":"papier","date":"2025-30-11"}],"error":false}}
`

**Comment by xirixiz** [comment]
Yeah, the thing is that the dates are sorted already for your area, for my area the dates aren't sorted. So I've added a sort function in the code to make sure to have the first upcoming trash date. Mijnafvalwijzer has a very weird implementation, but I`ll have a look at the code, maybe I can come up with some workaround for your issue.

**Comment by xirixiz** [comment]
Did some checking, but for most of the people a lot of dates are returned incorrect by mijnafvalwijzer. Best is to wait until it's fixed bij them. The thing is that a selection on the dates have to be made. So it's looking in the entries for the date of today, and then uses the data after that date to setup the sensors, so it's required actually that the dates match. What I did for now is that I modified the code so it only uses the first 25 entries from next_year. It seems that most of that data is at least correct. I`ll draft a new release.

```
return ophaaldagen_data + ophaaldagen_next_data[:25]
```
